### ISSUE TITLE
Bug in nova 2.9.0 - rulesForUpdate() method signature has changed

### ISSUE THREAD

**Issue by m-lotze** [eminiarts/nova-tabs#73]
In nova 2.9.0 the signature of the method `rulesForUpdate` has changed. this leads to an exception with this package:

```
Declaration of Eminiarts\Tabs\TabsOnEdit::rulesForUpdate(Laravel\Nova\Http\Requests\NovaRequest $request) 
should be compatible with
Laravel\Nova\Resource::rulesForUpdate(Laravel\Nova\Http\Requests\NovaRequest $request, $resource = NULL)
```

**Comment by LorenzoSapora** [comment]
Can confirm

**Comment by LorenzoSapora** [comment]
Update: rulesForUpdate (and, the other static functions in that trait) should align to nova's new arguments. Adding `$resource = null` to the block fixes the error, but unsure if that's the best solution.

```php
    /**
     * Get the validation rules for a resource update request.
     *
     * @param  \Laravel\Nova\Http\Requests\NovaRequest $request
     * @return array
     */
    public static function rulesForUpdate(NovaRequest $request, $resource = null)
    {
        return static::formatRules($request, (new static(static::newModel()))
                ->parentUpdateFields($request)
                ->mapWithKeys(function ($field) use ($request) {
                    return $field->getUpdateRules($request);
                })->all());
    }
```

I'd push a PR, but hopefully the maintainer (@bajramemini) can shed some light on this

**Comment by rikless** [comment]
confirmed

**Comment by vbezruchkin** [comment]
Another confirmation! I have the same problem on my Nova 2.9.0

**Comment by vbezruchkin** [comment]
Looks like there are many methods that are not compatible with the latest nova 2.9.0 :( I guess the package support might dropped as there is very little activity in terms of development. It's a sad story as the package is great itself!

**Comment by vincenzoraco** [comment]
@vbezruchkin Although your solution works perfectly fine (thank you for your time!), it is highly not recommended to change packages files just in order to fix problems. If the package is no longer maintained, we should either create a fork and make our changes on it or don't use the package at all.

If people don't want to fork it, I suggest to just avoid using the _TabsOnEdit_ trait for now until the package is updated with the fix.

**Comment by vbezruchkin** [comment]
Thanks for your feedback. That was just a temp solution for the guys, who had the same issues. Yes, I should have mentioned that in the original post :)

Regarding the fix - I've created a fork and have a pull request for the fix. To be honest, for me it's easier to update the vendor file with the working code, rather than edit around 70 models I have in my project and delete TabsOnEdit usage there :)

**Comment by vincenzoraco** [comment]
Wow! Indeed, 70 are a lot! I was thinking people used it for a few models. Anyway, let's hope that the maintainer push your edit through soon 👍 Thank you again for the time spent on it

**Comment by bajramemini** [comment]
Should be fixed in [1.2.1](https://github.com/eminiarts/nova-tabs/releases/tag/1.2.1) .
